src/cargo/core/compiler/mod.rs

```diff
@@ -764,6 +765,14 @@ fn add_error_format_and_color(
     } else {
         let mut color = true;
         match cx.bcx.build_config.message_format {
+            MessageFormat::Human if nightly_features_allowed() => {
+                if let (Some(width), _) | (_, Some(width)) = (
+                    cx.bcx.config.cli_unstable().terminal_width,
+                    cx.bcx.config.shell().accurate_err_width(),
+                ) {
+                    cmd.arg(format!("-Zterminal-width={}", width));
```

Comment:
Is there a reason this isn't included in the pipeline path?  Does it not affect the `rendered` field?  We plan on moving all interaction through the json path, so this may not work this way.

Reply:
I believe the `rustc` side ignores this codepath for the rendered field in json output, which is unfortunate for the integration here. We'll likely need some back and forth with the other repo to get both to be in the same page about this.